Marlin/src/module/stepper.cpp

```diff
@@ -2460,7 +2460,20 @@ hal_timer_t Stepper::block_phase_isr() {
 
         #if ENABLED(LIN_ADVANCE)
           if (la_active) {
-            const uint32_t la_step_rate = la_advance_steps < current_block->max_adv_steps ? current_block->la_advance_rate : 0;
```

Comment:
Why are these checks done? as long as we are in the acceleration phase, la_step_rate should always be `current_block->la_advance_rate`. What am I missing

Reply:
Is `la_advance_rate` chosen in a way which ensures that `max_adv_steps` can't be surpassed under any circumstances during the accel phase? Or is there a possibility for rounding errors or similar problems that would allow `la_advance_steps` to slightly overshoot `max_adv_steps`? This check looks like it was originally meant to catch errors like that. Don't know if it's still useful with this PR in place.

Reply:
Possibly. I assume it makes sure the exact added pressure is removed, although I believe this to have no effect in print quality (if anything it would result in a tiny bit of underextrusion since the extra pressure will likely have escaped the nozzle already somewhere else).

The idea of this PR is allowing for a less mathematically pedantic LA in favour of print speed, so if I understood correctly this check should indeed be irrelevant for it.

Reply:
I agree, small deviations on a single accel are unlikely to have any visible effect. But you still need to make sure that `la_advance_steps` returns to (precisely) zero in the following decel phase, otherwise the error will accummulate over multiple blocks.

Reply:
I think it shouldn't accumulate since la_step_rate is reset to zero on every cruise phase and at the beginning of each block. If the extruder ran a couple of steps too many, then the increased pressure will bleed out as over extrusion. 
as a sanity check I printed some benchies and I don't see any cumulative effects.

Reply:
do you have any bowden machine left? I'm really curious about how much faster this can run without starting to degrade quality. If acceleration is pushed beyond the extruder's best effort, this PR would start looking like k=0. And of course even if the extrusion were perfect, the increased acceleration can worsen other issues like ringing

Reply:
I've still got an Ender 3 Pro with Bowden setup, but that starts layer shifting above 6000 mm/s² on the Y axis (bed) as far as I remember my last tests. But I run `#define DEFAULT_EJERK 15.0` (originally even `20.0`) with `#define ADVANCE_K 0.45` and didn't have the impression that LA was the limiting factor for accel. According to your calculations in the first post, it should be, though.

Reply:
According to my calculations, LA should limit acceleration to 1004mm/s2.
Have you tried this PR? What I expect to happen is that the printer goes so fast that the extruder won't be able to fully keep up and it will result in less LA corrections applied. But reducing printing acc to say 3000 may be a good compromise. Also try increasing extruder acceleration as much as possible.
I'm super curious about any results you may get :)

Reply:
When I improved linear advance back in https://github.com/MarlinFirmware/Marlin/pull/24533 there was a lot of discussion about not causing cumulative errors in E steps. This check was part of preventing that. (IIRC the reasoning was this: Before my changes E steps were locked in sync with XYZ steps which gave this guarantee. After my changes there was the potential of rounding errors.)

I agree that this could be relaxed for PR which sacrifices exact step numbers for speed and lets the minor differences bleed out into the print pretty much invisibly.